Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,440 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+                                reflexivePair_obj_zero, reflexivePair_map_right,
+                                reflexivePair_map_reflexion, Category.assoc, h₃, reassoc_of% h₁]
+        | rightCompReflexion => dsimp
+                                simp only [Functor.map_comp, reflexivePair_obj_one,
+                                 reflexivePair_obj_zero, reflexivePair_map_left,
+                                 reflexivePair_map_reflexion, Category.assoc, h₃, reassoc_of% h₂])
+
+variable (F)
+
+/-- Every functor out of `WalkingReflexivePair` is isomorphic to the `reflexivePair` given by
+its components -/
+@[simps!]
+def diagramIsoReflexivePair :
+    F ≅ reflexivePair (F.map left) (F.map right) (F.map reflexion) :=
+  mkNatIso (Iso.refl _) (Iso.refl _)
+
+end NatTrans
+
+/-- A `reflexivePair` composed with a functor is isomorphic to the `reflexivePair` obtained by
+applying the functor at each map. -/
+@[simps!]
+def compRightIso {D : Type u₂} [Category.{v₂} D] {A B : C}
+    (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B) (sr : s ≫ g = 𝟙 B) (F : C ⥤ D) :
+    (reflexivePair f g s sl sr) ⋙ F ≅ reflexivePair (F.map f) (F.map g) (F.map s)
+      (by simp only [← Functor.map_comp, sl, Functor.map_id])
+      (by simp only [← Functor.map_comp, sr, Functor.map_id]) :=
+  mkNatIso (Iso.refl _) (Iso.refl _)
+
+lemma whiskerRightMkNatTrans {D : Type u₂} [Category.{v₂} D]
+    {F G: WalkingReflexivePair ⥤ C}
+    (u : F.obj zero ⟶ G.obj zero) (v : F.obj one ⟶ G.obj one)
+    {h₁ : F.map left ≫ u = v ≫ G.map left}
+    {h₂ : F.map right ≫ u = v ≫ G.map right}
+    {h₃ : F.map reflexion ≫ v = u ≫ G.map reflexion}
+    (H : C ⥤ D) :
+      (whiskerRight (mkNatTrans u v : F ⟶ G) H) = mkNatTrans
+            (H.map u) (H.map v)
+            (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₁])
+            (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₂])
+            (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₃]) := by
+  ext x; cases x <;> simp
+
+end
+
+/-- Any functor out of the WalkingReflexivePair yields a reflexive pair -/
+instance to_isReflexivePair {F : WalkingReflexivePair ⥤ C} :
+    IsReflexivePair (F.map .left) (F.map .right) :=
+  ⟨F.map .reflexion, map_reflexion_comp_map_left F, map_reflexion_comp_map_right F⟩
+
+end reflexivePair
+
+/-- A `ReflexiveCofork` is a cocone over a `WalkingReflexivePair`-shaped diagram. -/
+abbrev ReflexiveCofork (F : WalkingReflexivePair ⥤ C) := Cocone F
+
+namespace ReflexiveCofork
+
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+variable {F : WalkingReflexivePair ⥤ C}
+
+/-- The tail morphism of a reflexive cofork. -/
+abbrev π (G : ReflexiveCofork F) := G.ι.app zero
+
+lemma condition (G : ReflexiveCofork F) : F.map left ≫ G.π = F.map right ≫ G.π := by
+  erw [Cocone.w G left, Cocone.w G right]
+
+@[simp]
+lemma app_one_eq_π (G : ReflexiveCofork F) : G.ι.app zero = G.π := rfl
+
```

Comment:
It would be nice to add a constructor for `ReflexiveCofork`:
```lean
@[simps pt]
def mk {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
    ReflexiveCofork F where
  pt := X
  ι := reflexivePair.mkNatTrans π (F.map left ≫ π)

@[simp]
lemma mk_π {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
    (mk π h).π = π := rfl
```